src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Predicates/PredicateQuery.cs

```diff
@@ -60,8 +65,21 @@ public string GetColumnName(string propertyPath)
                 {
                     // Return the default alias
                     // ContentItemId -> ContentItemIndex.ContentItemId
-                    _usedAliases.Add(alias);
-                    return Dialect.QuoteForTableName($"{_tablePrefix}{alias}") + "." + Dialect.QuoteForColumnName(values[0]);
+                    // Ensures its actually a property of ContentItemIndex and gets the correct casing.
+                    var propertyName = values[0].ToLower();
+
+                    var indexKey = nameof(ContentItemIndex) + propertyName;
+
+                    if (!_indexMapping.TryGetValue(indexKey, out var columnName))
+                    {
+                        columnName = typeof(ContentItemIndex).GetProperty(propertyName, BindingFlags.Public | BindingFlags.Instance | BindingFlags.IgnoreCase)?.Name;
+                        _indexMapping.TryAdd(indexKey, columnName);
+                    }
+
+                    if (columnName != null) { 
```

Comment:
columnName is object

Reply:
```
if (columnName is object) {
```

Reply:
It will take more cycles for people to grasp what it means than the real perf benefits. And maybe one day we'll want to use the nullable instrcutions in c# and we will forget about these.

Reply:
yeah @sebastienros  it took @Jetski5822 sending me a link to read about it for me to grasp it haha